pkg/logs/agent.go

```diff
@@ -66,23 +67,34 @@ func NewAgent(sources *config.LogSources, services *service.Services, processing
 	// setup the pipeline provider that provides pairs of processor and sender
 	pipelineProvider := pipeline.NewProvider(config.NumberOfPipelines, auditor, diagnosticMessageReceiver, processingRules, endpoints, destinationsCtx)
 
+	// setup the launchers
+	lnchrs := launchers.NewLaunchers(sources, pipelineProvider, auditor)
```

Comment:
Is there an advantage to providing the launchers with their dependencies in `Start` vs in the initilizer? 
Not all launchers require a `regiestry` for example, and it moves the depednency requiremnt into the launcher interface instead of the implementation. 

I don't think this is a real problem, but from a code style perspective - i'd lean to providing dependences in one place (the implementaiton's initilizer) rather than making them part of the interface and thus having to inject 2 different sets of dependnecies into each launcher at different times.

Reply:
The idea is that things that "define" the launcher get passed to the constructor, while things that allow it to interact with other components get passed to Start().  The advantage is that this means an object can be created as an isolated component, then passed to another component (Agent, in this case) which supplies the specific bits for the interface (pipelineProvider, registry, and sources).

This makes sense for the scheduler, where for example serverless adds a purpose-specific scheduler.  It makes less sense for launchers, because they're all created in this function, but I decided to follow the same pattern.

This all feeds into the larger question of how components will be initialized and start up.  Would we rather have `cmd/serverless` create an empty logs.Agent instance and add subcomponents (AddScheduler/AddLauncher) as necessary?  Or have `logs.start` get an indication of what flavor is running, and map that to a set of subcomponents internally?  Hopefully the component project will sort this out, at which time we can refactor this stuff to match.

Reply:
Got it. Thanks for the explanation.